tools/squidclient/squidclient.cc

```diff
@@ -512,8 +510,13 @@ main(int argc, char *argv[])
         }
         if (ProxyAuthorization.user)
             ProxyAuthorization.commit(msg);
-        if (OriginAuthorization.user)
+        if (OriginAuthorization.user) {
+            const auto savedPassword = OriginAuthorization.password;
+            if (pathPassword)
+                OriginAuthorization.password = pathPassword;
```

Comment:
I adjusted the code to abort if either pathPassword or -w/-W have been specified without specifying a username. 

>  We can add pathPassword as an additional (ORed) condition 

We could do this but the old code actually never worked with a missing username: i.e., *Authorization headers were not created if just a password had been provided. We could set username to an empty string in this case (and generate the value from the ':password' string - but this would change the old behavior.